src/hooks/screen/useChatScreenHook.ts

```diff
@@ -124,17 +127,62 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       }
 
       let response;
+
+      const allAttachmentPromises: Promise<void>[] = attachments.map(async (item: any) => {
+        if (item.type === 'image') {
+          const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+          return {
+            ...item,
+            asset_url: uploadedImageUrl.data.url,
+            thumb_url: uploadedImageUrl.data.url
+          };
+        }
+        if (item.type === 'video') {
+          const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+          const uploadedUrl = await ImageUtils.uploadFile(
+            item.video_path,
+            item.video_name,
+            item.video_type
+          );
+          return {
+            ...item,
+            asset_url: uploadedImageUrl.data.url,
+            thumb_url: uploadedImageUrl.data.url,
+            video_path: uploadedUrl.data.url
+          };
+        }
+        if (item.type === 'file') {
+          const uploadedUrl = await ImageUtils.uploadFile(
+            item.file_path,
+            item.file_name,
+            item.file_type
+          );
+          return {
+            ...item,
+            asset_url: uploadedUrl.data.url,
+            thumb_url: uploadedUrl.data.url,
+            file_path: uploadedUrl.data.url
+          };
+        }
+
+        return item;
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code seems to handle the uploading of different types of attachments (image, video, file). However, there's no error handling in case any of the upload operations fail. It would be good to add try-catch blocks around the upload operations and handle errors appropriately. Also, the same `ImageUtils.uploadImage(item.asset_url)` is used for both image and video types which may not be correct for videos.

```diff
  const allAttachmentPromises: Promise<void>[] = attachments.map(async (item: any) => {
    try {
      if (item.type === 'image') {
        const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
        return {
          ...item,
          asset_url: uploadedImageUrl.data.url,
          thumb_url: uploadedImageUrl.data.url
        };
      }
      if (item.type === 'video') {
-       const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+       const uploadedVideoThumbUrl = await ImageUtils.uploadVideoThumbnail(item.asset_url);
        const uploadedUrl = await ImageUtils.uploadFile(
          item.video_path,
          item.video_name,
          item.video_type
        );
        return {
          ...item,
          asset_url: uploadedVideoThumbUrl.data.url,
          thumb_url: uploadedVideoThumbUrl.data.url,
          video_path: uploadedUrl.data.url
        };
      }
      if (item.type === 'file') {
        const uploadedUrl = await ImageUtils.uploadFile(
          item.file_path,
          item.file_name,
          item.file_type
        );
        return {
          ...item,
          asset_url: uploadedUrl.data.url,
          thumb_url: uploadedUrl.data.url,
          file_path: uploadedUrl.data.url
        };
      }
    } catch (error) {
      console.error(`Error uploading ${item.type}: `, error);
      throw error;
    }

    return item;
  });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->